### PR TITLE
Provide compatibility with WaaS 1.4.0

### LINKED ISSUE #21: Provide compatibility with WaaS 1.4.0
The SDK features should be updated to the recently [released](https://tangany.com/blog/release-notes-1-4/) version 1.4.0 of WaaS. These include: 
- Asynchronous requests (execution and status request)
- Universal Ethereum Smart Contract writing
- Signing endpoints
- Various changes like gas tank or custom gas price for Ethereum transactions

### PR DESCRIPTION
**Pull request by adrianmitter**
Resolves #21

### CODE REVIEW

File: src/blockchain-wallet.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import * as t from "typeforce";
+import {Waas} from "./waas";
+import {IWaasMethod} from "./waas-method";
+import {Wallet} from "./wallet";
+
+/**
+ * Base for classes that provide an interface for wallet actions, e.g. on the Bitcoin or Ethereum blockchain.
+ */
+export abstract class BlockchainWallet implements IWaasMethod {
```
Comment (adrianmitter) [inline src/blockchain-wallet.ts:9]: I'm not quite sure if this is an appropriate class name or if it's confusing.

File: src/blockchain-wallet.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return this.walletInstance.wallet;
+    }
+
+    /**
+     * Determines the request id from an server response to an asynchronous request.
+     * The server always returns the location of the server resource that represents the status of the
+     * asynchronous request. From this, only the ID is extracted.
+     * @param serverResponse - Object that represents the raw server response of an asynchronous endpoint
+     */
+    protected extractRequestId(serverResponse: { [key: string]: any }): string {
```
Comment (adrianmitter) [inline src/blockchain-wallet.ts:25]: Is this actually the right place for this method? Or would it be better in a separate Util class?

File: src/waas.ts [hunk 3]
```diff
@@ -92,10 +95,10 @@ export interface IWaitForTxStatus {
  * @param options.clientSecret - Subscription client secret
  * @param options.subscription - Subscription code
  * @param options.vaultUrl - Tangany vault url
- * @param options.ethereumNetwork - Public Ethereum network name (@see https://tangany.docs.stoplight.io/api/models/ethereum-public-network) or private Ethereum network url (@see https://tangany.docs.stoplight.io/api/models/ethereum-private-network)
+ * @param options.ethereumNetwork - Public Ethereum network name or private Ethereum network url
```
Comment (adrianmitter) [inline src/waas.ts:98]: In the new documentation there is no longer a section for the models like Public Ethereum Network or similar. Therefore no link can be inserted here. Should we then give hardcoded examples?

File: readme.md [hunk 4]
```diff
@@ -196,7 +190,7 @@ DEBUG=waas-js-sdk:*
 ```
 
 ## API documentation
-Full API documentation is available at https://tangany.docs.stoplight.io/
+Full API documentation is available at https://docs.tangany.com
 
 ***
 <div align="center">
```
Comment (kepikoi) [inline readme.md:196]: Please update the "Available configuration headers" table with new configuration headers:
- useGasTank
- ethereumTxFee

File: src/blockchain-wallet.ts (hunk 1 above)
Comment (kepikoi) [reply src/blockchain-wallet.ts:9]: I think it is appropriate

File: src/waas.ts (hunk 3 above)
Comment (kepikoi) [reply src/waas.ts:98]: I would just omit the link and refrain from examples like in the other options. The readme file provides at least one example for private Ethereum network usage

File: test/async-requests.e2e.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		let txRequest;
+
+		before(async function () {
+			// Get the Ethereum-specific address of the wallet
+			// Execute this at the beginning of the test suite and not in a test because it is not a meaningful step of this workflow (for asynchronous requests)
+			const { address } = await api.wallet(createdWallet).eth().get();
+			createdWalletEthAddress = address;
+		});
+
+		it("should start an asynchronous request that sends a few tokens", async function () {
```
Comment (kepikoi) [inline test/async-requests.e2e.js:50]: "...sends some ether"

File: readme.md (hunk 4 above)
Comment (kepikoi) [reply readme.md:196]: Also please add some code examples for the new endpoints to readme.md

File: test/async-requests.e2e.js (hunk 5 above)
Comment (adrianmitter) [reply test/async-requests.e2e.js:50]: This is of course correct as in this test no tokens are sent, but ether.

**Pull request merged**